### ISSUE TITLE
New data source to list organization OUs contained within an AWS organization/OU

### ISSUE THREAD

**Issue by mungoknotwise** [terraform-providers/terraform-provider-aws#9884]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It would be very useful to have a data resource to list all organizational units under a parent organizational unit with root being the top-level parent.

### New or Affected Resource(s)

* data.aws_organizations_organizational_unit

### Potential Terraform Configuration

```hcl
data "aws_organizations_organization" "org" {}

data "aws_organizations_organizational_unit" "top_level_ous" {
  parent_id = org.roots[0].id
}
```

### References

* #0000

**Comment by DavidFluck** [comment]
I'm interested in working on this (I actually already have something half working). I should be able to submit a pull request within the next week or two.

**Comment by dbektas** [comment]
Can this PR be checked? It is very useful data provider for AWS organization service that is currently sadly missing.

**Comment by silvestriluca** [comment]
@bflad The PR [#10395](https://github.com/terraform-providers/terraform-provider-aws/pull/10395) has recently been fixed and is ready for review. Could you please review it and, if ok, move it forward to have the PR merged ? We have already use cases for it in near production stage.
Thanks.

**Comment by bflad** [comment]
Support for a new `aws_organizations_organizational_units` data source has been merged and will release with version 2.43.0 of the Terraform AWS Provider, tomorrow. Thanks to @DavidFluck for the implementation. 👍
